lib/rules/no-implied-eval.js

```diff
@@ -12,51 +12,78 @@
     //--------------------------------------------------------------------------
 
     /**
-     * Checks if the first argument of a given CallExpression node is a string literal.
-     * @param {ASTNode} node The CallExpression node the check.
-     * @returns {boolean} True if the first argument is a string literal, false if not.
+     * Get the last element of an array, without modifying arr, like pop(), but non-destructive.
+     * @param {array} arr What to inspect
+     * @returns {*} The last element of arr
+     * @private
      */
-    function hasStringLiteralArgument(node) {
-        var firstArgument = node.arguments[0];
-
-        return firstArgument && firstArgument.type === "Literal" && typeof firstArgument.value === "string";
+    function last(arr) {
+        return arr ? arr[arr.length - 1] : null;
     }
 
     /**
-     * Checks if the given MemberExpression node is window.setTimeout or window.setInterval.
+     * Checks if the given MemberExpression node is a dangerous identifier on window.
      * @param {ASTNode} node The MemberExpression node to check.
-     * @returns {boolean} Whether or not the given node is window.set*.
+     * @returns {boolean} Whether or not the given node is potentially dangerous.
+     * @private
      */
-    function isSetMemberExpression(node) {
+    function isDangerousMemberExpression(node) {
         var object = node.object,
             property = node.property,
-            hasSetPropertyName = IMPLIED_EVAL.test(property.name) || IMPLIED_EVAL.test(property.value);
-
-        return object.name === "window" && hasSetPropertyName;
+            hasDangerousName = CALLEE_RE.test(property.name) || CALLEE_RE.test(property.value);
 
+        return object.name === "window" && hasDangerousName;
     }
 
     /**
-     * Determines if a node represents a call to setTimeout/setInterval with
+     * Determines if a node represents a call to a potentially dangerous method
      * a string argument.
-     * @param {ASTNode} node The node to check.
+     * @param {ASTNode} node The CallExpression to check.
      * @returns {boolean} True if the node matches, false if not.
      * @private
      */
-    function isImpliedEval(node) {
+    function isDangerousCallExpression(node) {
         var isMemberExpression = (node.callee.type === "MemberExpression"),
             isIdentifier = (node.callee.type === "Identifier"),
-            isSetMethod = (isIdentifier && IMPLIED_EVAL.test(node.callee.name)) ||
-                (isMemberExpression && isSetMemberExpression(node.callee));
+            isDangerousCallee =
+                (isIdentifier && CALLEE_RE.test(node.callee.name)) ||
+                (isMemberExpression && isDangerousMemberExpression(node.callee));
+
+        return isDangerousCallee && node.arguments.length;
+    }
 
-        return isSetMethod && hasStringLiteralArgument(node);
+    function hasDangerousParent(node) {
+        var parent = last(context.getAncestors(node));
```

Comment:
@bgw I don't remember the whole history, but `node.parent` was added later then `context.getAncestors` as far as I recall. In a lot of rules you also need more then just immediate parent, so people got used to `context.getAncestors`. Actually we were talking to too long ago about removing `node.paren` because it's underused and can be replicated with other functionality, but we decided against it anyways.